jgiven-html5-report/src/main/java/com/tngtech/jgiven/report/html5/Html5AttachmentGenerator.java

```diff
@@ -195,4 +201,78 @@ static BufferedImage scaleDown( BufferedImage before ) {
         }
         return imageArray;
     }
+
+    private void writeThumbnailForSVG(File initialSVG, File thumbnailSVG) {
+        String base64PNGImage = getPNGFromSVG(initialSVG);
+        byte[] scaledDownInBytes = compressToThumbnail(base64PNGImage, "png");
+        Dimension imageDimension = getImageDimension(scaledDownInBytes);
+        String base64ScaledDownContent = BaseEncoding.base64().encode(scaledDownInBytes);
+        createSVGThumbFile(thumbnailSVG, base64ScaledDownContent, imageDimension);
+    }
+
+    Dimension getImageDimension(byte[] givenImage) {
+        ByteArrayInputStream bais = new ByteArrayInputStream(givenImage);
+        Dimension dimension = new Dimension();
+        try {
+            BufferedImage image = ImageIO.read(bais);
+            dimension.height = image.getHeight();
+            dimension.width = image.getWidth();
+        } catch (IOException e) {
+            log.error("The converted png image cannot be read.");
+        }
+
+        return dimension;
+    }
+
+    void createSVGThumbFile(File targetFile, String base64Image, Dimension dimension) {
+        String xmlFormat = getXMLFormat(base64Image, dimension);
+        try {
+            FileWriter fileWriter = new FileWriter(targetFile);
+            fileWriter.write(xmlFormat);
+            fileWriter.close();
+        } catch (IOException e) {
+            log.error("Error writing the thumbnail svg to " + targetFile);
+        }
+    }
+
+    private String getXMLFormat(String base64Image, Dimension dimension) {
+        return "<svg xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" " +
+                "height=\"" + dimension.getHeight() + "px\" width=\"" + dimension.getWidth() + "px\">"+
+                "<image height=\"100%\" width=\"100%\" xlink:href=\"data:image/png;base64, " + base64Image + "\"/>" +
+                "</svg>";
+    }
+
+    String getPNGFromSVG(File givenSVG) {
+        PNGTranscoder transcoder = new PNGTranscoder();
+        TranscoderInput transcoderInput = new TranscoderInput();
+        TranscoderOutput transcoderOutput = new TranscoderOutput();
+
+        FileInputStream fis = null;
+        try {
+            fis = new FileInputStream(givenSVG);
+        } catch (FileNotFoundException e) {
+            log.error("Error while reading the initial svg file.");
+            return null;
+        }
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+
+        transcoderInput.setInputStream(fis);
+        transcoderOutput.setOutputStream(baos);
+
+        try {
+            transcoder.transcode(transcoderInput, transcoderOutput);
+        } catch (TranscoderException e) {
+            e.printStackTrace();
+            log.error("Error while transcoding the svg file to png. Is the svg formatted correctly?");
+            return null;
+        }
+
+        try {
+            fis.close();
```

Comment:
Could you use a try-with-resources here? You could catch separate exceptions if you are concerned about logging the right exception and messages.